src/tools/fuzzing/fuzzing.cpp

```diff
@@ -2323,6 +2323,13 @@ Expression* TranslateToFuzzReader::makeCompoundRef(Type type) {
   // least avoids infinite recursion here, and we emit a valid (but not that
   // useful) wasm.
   if (type.isNonNullable() && (random.finished() || nesting >= LIMIT)) {
+    // If we have a function context then we can at least emit a local.get,
+    // perhaps, which is less bad. Note that we need to check typeLocals
+    // manually here to avoid infinite recursion (as makeLocalGet will fall back
+    // to us, if there is no local).
+    if (funcContext && !funcContext->typeLocals[type].empty()) {
```

Comment:
Does this look for locals holding subtypes as well? If not, it could.

Reply:
Good point, it doesn't. I added a TODO.